core/src/main/java/org/apache/calcite/rex/RexSimplify.java

```diff
@@ -1887,11 +1887,10 @@ private static <C extends Comparable<C>> RangeSet<C> residue(RexNode ref,
       case GREATER_THAN:
       case GREATER_THAN_OR_EQUAL:
         final RexCall call = (RexCall) predicate;
-        if (call.operands.get(0).equals(ref)
-            && call.operands.get(1) instanceof RexLiteral) {
-          final RexLiteral literal = (RexLiteral) call.operands.get(1);
-          final C c1 = literal.getValueAs(clazz);
-          assert c1 != null : "value must not be null in " + literal;
+        final Comparison comparison = Comparison.of(call);
```

Comment:
We change here because previous logic only checks `ref > 1`, not `1 < ref`, we use `Comparison` to refine this part of code, and support the reverse order comparison.